src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -231,26 +454,12 @@ internal static WwwAuthenticateParameters CreateWwwAuthenticateParameters(IDicti
                 wwwAuthenticateParameters.Error = value;
             }
 
-            return wwwAuthenticateParameters;
-        }
-
-        /// <summary>
-        /// Extracts a key value pair from an expression of the form a=b
-        /// </summary>
-        /// <param name="assignment">assignment</param>
-        /// <returns>Key Value pair</returns>
-        private static KeyValuePair<string, string> ExtractKeyValuePair(string assignment)
-        {
-            string[] segments = CoreHelpers.SplitWithQuotes(assignment, '=')
-                .Select(s => s.Trim().Trim('"'))
-                .ToArray();
-
-            if (segments.Length != 2)
+            if (values.TryGetValue("nonce", out value))
             {
-                throw new ArgumentException(nameof(assignment), $"{assignment} isn't of the form a=b");
+                wwwAuthenticateParameters.PopNonce = value.Replace("\"", string.Empty);
```

Comment:
Why do you replace " character?

Reply:
I think this was some left-over logic from a previous implementation. removing